example/kokkos_pi/kokkos_pi.cpp

```diff
@@ -0,0 +1,381 @@
+  // Initialize the timer and test
+  Kokkos::Timer timer;
+
+  for (int i_run = 0; i_run < n_burn + n_perf; i_run++) {
+    if (i_run == n_burn) {
+      // Burn in time is over, start timing
+      Kokkos::fence();
+      timer.reset();
+    }
+
+    // Run the function timing performance
+    perf_func();
+  }
+
+  // Time it
+  Kokkos::fence();
+  double perf_time = timer.seconds();
+
+  return perf_time;
+}
+
+static void usage(std::string program) {
+  std::cout << std::endl
+            << "    Usage: " << program << " n_block n_mesh n_iter" << std::endl
+            << std::endl
+            << "             n_block = size of each mesh block on each axis" << std::endl
+            << "              n_mesh = number mesh blocks alogn each axis" << std::endl
+            << "              n_iter = number of iterations to time" << std::endl
+            << "            [Radius] = Optional: Radius of sphere" << std::endl
+            << "                                 Defaults to 1.0" << std::endl
+            << std::endl;
+}
+
+static double sumArray(MeshBlock *firstBlock, const int &n_block) {
+  // This policy is over one block
+  const int n_block2 = n_block * n_block;
+  const int n_block3 = n_block * n_block * n_block;
+  auto policyBlock = Kokkos::RangePolicy<>(Kokkos::DefaultExecutionSpace(), 0, n_block3,
+                                           Kokkos::ChunkSize(512));
+  double theSum = 0.0;
+  // reduce the sum on the device
+  // I'm pretty sure I can do this better, but not worried about performance for this
+  MeshBlock *pmb = firstBlock;
+  while (pmb) {
+    Container<Real> &base = pmb->real_containers.Get();
+    auto inOrOut = base.PackVariables({Metadata::Independent});
+    double oneSum;
+    Kokkos::parallel_reduce(
+        "Reduce Sum", policyBlock,
+        KOKKOS_LAMBDA(const int &idx, double &mySum) {
+          const int k_grid = idx / n_block2;
+          const int j_grid = (idx - k_grid * n_block2) / n_block;
+          const int i_grid = idx - k_grid * n_block2 - j_grid * n_block;
+          mySum += inOrOut(0, k_grid + NGHOST, j_grid + NGHOST, i_grid + NGHOST);
+        },
+        oneSum);
+    Kokkos::fence();
+    theSum += oneSum;
+    pmb = pmb->next;
+  }
+  // calculate Pi
+  return theSum;
+}
+
+static MeshBlock *setupMesh(const int &n_block, const int &n_mesh, const double &radius,
+                            View2D &xyz, const int NG = 0) {
+  // *** Kludge warning ***
+  // Since our mesh is not GPU friendly we set up a hacked up
+  // collection of mesh blocks.  The hope is that when our mesh is
+  // up to par we will replace this code with the mesh
+  // infrastructure.
+
+  const Real dxyzCell = radius / static_cast<Real>(n_mesh * n_block);
+  auto h_xyz = Kokkos::create_mirror_view(xyz);
+
+  // Set up our mesh.
+  Metadata myMetadata({Metadata::Independent, Metadata::Cell});
+  MeshBlock *firstBlock = nullptr;
+  MeshBlock *lastBlock = nullptr;
```

Comment:
some day we should reconsider using raw pointers for meshblocks. Probably shouldn't be today though.